splash/tests/test_execute.py

```diff
@@ -1809,3 +1811,160 @@ def test_get_perf_stats(self):
         now = time.time()
         self.assertLess(now - 120, out['walltime'])
         self.assertLess(out['walltime'], now)
+
+
+class WindowSizeTest(BaseLuaRenderTest):
+    """This is a test for window & viewport size interaction in Lua scripts."""
+
+    BASIC_SCRIPT = """
+function get_dims(splash)
+    return {
+        inner = splash:evaljs("window.innerWidth") .. "x" .. splash:evaljs("window.innerHeight"),
+        outer = splash:evaljs("window.outerWidth") .. "x" .. splash:evaljs("window.outerHeight"),
+        client = (splash:evaljs("document.documentElement.clientWidth") .. "x"
+                  .. splash:evaljs("document.documentElement.clientHeight"))
+    }
+end
+
+function main(splash)
+    alter_state(splash)
+    return get_dims(splash)
+end
+
+"""
+
+    def run_dims_test(self, alter_fn, **kwargs):
+        script = dedent(self.BASIC_SCRIPT) + "\n" + dedent(alter_fn)
+        resp = self.request_lua(script, kwargs)
+        if resp.ok:
```

Comment:
This raises a catchable RuntimeError which is checked against in some tests. Although, I suppose I could use `resp.json` in successful code path.